### PR TITLE
[#21] Add aggregate

### PR DESCRIPTION
**Pull request by willbasky**
Resolve [#21](https://github.com/kowainik/hintman/issues/21)

### CODE REVIEW

File: src/Hintman/Suggestion/Aggregation.hs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+       ) where
+
+import Hintman.Suggestion.Core (Line (..), Suggestion (..), toLines)
+
+import qualified Hintman.Suggestion.TrailingNewline as STN
+import qualified Hintman.Suggestion.TrailingSpaces as STS
+
+
+aggregate :: FilePath -> Text -> [Suggestion]
+aggregate path txt = concat [deleteSuggest, editSuggest]
```
Comment (sphaso) [inline src/Hintman/Suggestion/Aggregation.hs]: we're not dealing with different suggestions for the same line for now? /cc @ChShersh

File: src/Hintman/Suggestion/Aggregation.hs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  where
+    -- First, `aggregate deletes trailingNewLines
+    deleteSuggest :: [Suggestion]
+    deleteSuggest = STN.suggest path lineList
+
+    deletedLines :: [Line]
+    deletedLines = map suggestionLine deleteSuggest
+    -- Second, it consumes the rest lines to suggest edit.
+    toEditLines :: [Line]
+    toEditLines = filter (\l -> l `notElem` deletedLines) lineList
```
Comment (cronokirby) [inline src/Hintman/Suggestion/Aggregation.hs]: Why not `not
```suggestion
    toEditLines = filter (`notElem` deletedLines) lineList
```

File: src/Hintman/Suggestion/Aggregation.hs (hunk 1 above)
Comment (ChShersh) [reply src/Hintman/Suggestion/Aggregation.hs]: @sphaso Yes, not now. Will solve the problem under different issue.

File: src/Hintman/Suggestion/Aggregation.hs [hunk 3]
```diff
@@ -0,0 +1,28 @@
+module Hintman.Suggestion.Aggregation
```
Comment (vrom911) [inline src/Hintman/Suggestion/Aggregation.hs]: Heh, now looking at this over time I really feel that the top-module documentation is missing here..

File: src/Hintman/Suggestion/Aggregation.hs [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+       ( aggregate
+       ) where
+
+import Hintman.Suggestion.Core (Line (..), Suggestion (..), toLines)
+
+import qualified Hintman.Suggestion.TrailingNewline as STN
+import qualified Hintman.Suggestion.TrailingSpaces as STS
+
+
+aggregate :: FilePath -> Text -> [Suggestion]
```
Comment (vrom911) [inline src/Hintman/Suggestion/Aggregation.hs]: Add function documentation as well.

File: src/Hintman/Suggestion/Core.hs [hunk 5]
```diff
@@ -24,7 +22,7 @@ data Suggestion = Suggestion
 data LineChange = LineChange
     { lineChangeNew     :: Text
     , lineChangeComment :: Maybe Text
-    , lineChangeType    :: ChangeType
+    , lineChangeType    :: SuggestionType
```
Comment (vrom911) [inline src/Hintman/Suggestion/Core.hs]: Mm, this is semantically different from what was here before. `ChangeType` is not that important in this case?

**Comment by ChShersh** [comment]
@willbasky You can update on latest `master` to get newer versions of dependencies.
Also, do you still have time and plans to work on this issue?

**Comment by willbasky** [comment]
@ChShersh ops, I missed last review. Yes, I will finished it.

File: src/Hintman/Suggestion/Core.hs (hunk 5 above)
Comment (willbasky) [reply src/Hintman/Suggestion/Core.hs]: I guess it was swapped because suggestion is more general then _change_, and can be used for _delete_.

**Pull request merged**